internal/publish/set_env_vars.go

```diff
@@ -31,7 +34,17 @@ func (p *defaultPublisher) setEnvVars(
 	for name, value := range env {
 		log.Info("Setting environment variable", "name", name, "value", value)
 	}
-	err := client.SetEnvVars(contentID, env, log)
+
+	for name := range secrets {
+		log.Info("Setting secret as environment variable", "name", name)
+	}
+
+	// Combine env and secrets into one environment for Connect
+	combinedEnv := make(map[string]string)
+	maps.Copy(combinedEnv, env)
+	maps.Copy(combinedEnv, secrets)
+
+	err := client.SetEnvVars(contentID, combinedEnv, log)
```

Comment:
same here, would be good to have tests for these changes

Reply:
After some pairing I will get something pushed up momentarily.

Reply:
Added tests in https://github.com/posit-dev/publisher/pull/2273/commits/e085f8b84703f1376f32c7e3afe1918ba36ddcb0